compiler/rustc_middle/src/thir.rs

```diff
@@ -748,6 +749,21 @@ pub enum PatKind<'tcx> {
         value: mir::Const<'tcx>,
     },
 
+    /// Pattern lowered from an inline constant
```

Comment:
Maybe something like "Inline constant found while lowering pattern"? To be more accurate wrt what we do with ranges